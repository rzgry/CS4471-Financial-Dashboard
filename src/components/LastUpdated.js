import React from 'react';

const LastUpdated = ({ timestamp }) => (
  <div style={{ margin: '1em', marginBottom: '2em' }}>
    {timestamp && (
      <p style={{ float: 'right', color: 'gray' }}>
        {`Last updated: ${new Date(timestamp).toLocaleString()}`}
      </p>
    )}
  </div>
);

export default LastUpdated;
