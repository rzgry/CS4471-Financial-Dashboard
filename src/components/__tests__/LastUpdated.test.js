import React from 'react';
import renderer from 'react-test-renderer';
import LastUpdated from '../LastUpdated';

describe('<LastUpdated />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<LastUpdated timestamp={new Date().getTime()} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
