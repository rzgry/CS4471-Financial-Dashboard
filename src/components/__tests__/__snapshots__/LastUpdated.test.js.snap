// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LastUpdated /> renders correctly 1`] = `
<div
  style={
    Object {
      "margin": "1em",
      "marginBottom": "2em",
    }
  }
>
  <p
    style={
      Object {
        "color": "gray",
        "float": "right",
      }
    }
  >
    Last updated: 2019-04-03, 6:09:21 p.m.
  </p>
</div>
`;
